src/main/java/org/spongepowered/common/data/processor/data/entity/IgnoreSleepingDataProcessor.java

```diff
@@ -0,0 +1,57 @@
+package org.spongepowered.common.data.processor.data.entity;
+
+import net.minecraft.entity.player.EntityPlayerMP;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableIgnoreSleepingData;
+import org.spongepowered.api.data.manipulator.mutable.entity.IgnoreSleepingData;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeIgnoreSleepingData;
+import org.spongepowered.common.data.processor.common.AbstractEntitySingleDataProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+import org.spongepowered.common.mixin.core.entity.player.MixinEntityPlayerMP;
+
+import java.util.Optional;
+
+public class IgnoreSleepingDataProcessor extends
+        AbstractEntitySingleDataProcessor<EntityPlayerMP, Boolean, Value<Boolean>, IgnoreSleepingData, ImmutableIgnoreSleepingData> {
+
+    public IgnoreSleepingDataProcessor() {
+        super(EntityPlayerMP.class, Keys.IGNORE_SLEEPING);
+    }
+
+    @Override
+    public DataTransactionResult removeFrom(ValueContainer<?> container) {
+        return DataTransactionResult.failNoData();
+    }
+
+    @Override
+    protected boolean set(EntityPlayerMP dataHolder, Boolean value) {
+        ((MixinEntityPlayerMP) (Object) dataHolder).setSleepingIgnored(value);
```

Comment:
Could you move this method to `IMixinEntityPlayerMP`? Mixin classes can't be referenced directly from non-mixin code.

Reply:
Actually, this seems to be available in `Player` directly.